lib/config/presets/index.ts

```diff
@@ -218,15 +219,22 @@ export async function getPreset(
   }
   const { presetSource, repo, presetPath, presetName, tag, params } =
     parsePreset(preset);
-  let presetConfig = await presetSources[presetSource].getPreset({
-    repo,
-    presetPath,
-    presetName,
-    tag,
-  });
+  const cacheKey = `preset:${preset}`;
+  const cachedResult = memCache.get<Preset>(cacheKey);
+  let presetConfig = is.undefined(cachedResult)
+    ? await presetSources[presetSource].getPreset({
+        repo,
+        presetPath,
+        presetName,
+        tag,
+      })
+    : cachedResult;
   if (!presetConfig) {
     throw new Error(PRESET_DEP_NOT_FOUND);
   }
+  if (is.undefined(cachedResult)) {
+    memCache.set(cacheKey, presetConfig);
+  }
```

Comment:
Why only cache undefined?

Reply:
Before we were checking if this is null or undefined. I will change to a simple if statement now

Reply:
If the previous cache result was undefined (not found) then it caches the new result

Reply:
@viceice Does this look better or keep what we have ?
```  
const { presetSource, repo, presetPath, presetName, tag, params } =
    parsePreset(preset);
  const cacheKey = `preset:${preset}`;
  let presetConfig: Preset | null | undefined = memCache.get<Preset>(cacheKey);
  if (is.nullOrUndefined(presetConfig)) {
    presetConfig = await presetSources[presetSource].getPreset({
      repo,
      presetPath,
      presetName,
      tag,
    });
    if (presetConfig) {
      memCache.set(cacheKey, presetConfig);
    }
  }
  if (!presetConfig) {
    throw new Error(PRESET_DEP_NOT_FOUND);
  }
  logger.trace({ presetConfig }, `Found preset ${preset}`);```

Reply:
updated my suggestion